utils/network.py

```diff
@@ -0,0 +1,112 @@
+        """
+        erp_session = cls(session=requests.Session())
+        erp_session.session_token, erp_session.sso_token = erp.login(erp_session.headers, erp_session.session)
+        erp_session.roll_number = erp.ROLL_NUMBER
+        erp_session.refresh_cookies()
+        return erp_session
+
+    def check_login(self) -> bool:
+        r"""
+        Checks if proper tokens and cookies exist
+        """
+        return self.sso_token != None and self.cookie != None and self.session != None
+
+    def post(self, url: str, data=None, json=None, cookies=None) -> Response:
+        """
+        Wrapper for post
+        Args:
+            url: URL for post request
+            data: (optional) Dictionary, list of tuples, bytes, or file-like object to send in the body of the Request
+            json: (optional) json to send in the body of the Request.
+            cookies: (optional) set to True to use stored cookies.
+
+        Returns: Response
+        """
+        if cookies is True:
+            cookies = self.cookie
+        return self.session.post(url=url, data=data, json=json, headers=self.headers, cookies=cookies)
+
+    def refresh_cookies(self) -> dict[str, str]:
+        r"""
+            Refreshes cookies and returns them for further use
+        """
+        # This is just a hack to get cookies. TODO: do the standard thing here
+        timetable_details = {
+            "ssoToken": self.sso_token,
+            "module_id": '16',
+            "menu_id": '40',
+        }
+        self.post(self.ERP_TIMETABLE_URL, data=timetable_details)
+        cookie_val = None
+        for a in self.session.cookies:
+            if a.path == "/Acad/":
+                cookie_val = a.value
+
+        cookie = {
+            'JSESSIONID': cookie_val,
+        }
+        self.cookie = cookie
+        return cookie
+
+    def get_timetable_details(self) -> dict[str, str]:
+        return {
+            "ssoToken": self.sso_token,
+            "module_id": '16',
+            "menu_id": '40',
+        }
+
+    def get_course_page_details(self) -> dict[str, str]:
+        if SEM_BEGIN.month > 6:
+            # autumn semester
+            sem_number = (int(SEM_BEGIN.strftime("%y")) - int(self.roll_number[:2])) * 2 + 1
+        else:
+            # spring semester
+            sem_number = (int(SEM_BEGIN.strftime("%y")) - int(self.roll_number[:2])) + 2
+
+        return {
+            "ssoToken": self.sso_token,
+            "semno": sem_number,
+            "rollno": self.roll_number,
+            "order": "asc"
+        }
+
+    def get_course_names(self) -> dict[str, str]:
+        r"""
+            Returns a mapping of course codes to course names
+        """
+        r = self.post(self.COURSES_URL, data=self.get_course_page_details())
+        sub_dict = {item["subno"]: item["subname"] for item in r.json()}
+        return {k: v.replace("&amp;", "&") for k, v in sub_dict.items()}
+
+
+if __name__ == "__main__":
+    ERPSession.login()
```

Comment:
Is this required, since we are calling `ERPSession.login()` anyways in `gyft.py` ?